### PR TITLE
Prevent attacker from manipulating FPU tag word used in SGX enclave

### PR DESCRIPTION
**Pull request by raoulstrackx**
Insufficient sanitization of the x87 FPU tag word in the trusted enclave runtime allowed unprivileged adversaries in the containing host application to induce incoherent or unexpected results for ABI-compliant compiled enclave application code that uses the x87 FPU.

Vulnerability was disclosed to us by Fritz Alder, Jo Van Bulck, David Oswald and Frank Piessens

cc: @jethrogb

### CODE REVIEW

**Comment by jethrogb** [comment]
These lines can be deleted now, I think?

https://github.com/rust-lang/rust/pull/73471/files#diff-e36fb308beaa14873467f172667231fbR191-R193

**Comment by raoulstrackx** [comment]
Those two lines (and mistakenly the two labels/values they refer to) were exactly the ones I was looking at before. I shouldn't have let you convinced me they were needed. :) Let me check again.

**Comment by jethrogb** [comment]
I thought you were talking about https://github.com/rust-lang/rust/blob/daedb7920f48941bd8ffa1b1463b417b1641c823/src/libstd/sys/sgx/abi/entry.S#L321-L322

**Comment by raoulstrackx** [comment]
Yes I mixed the two up.

File: src/libstd/sys/sgx/abi/entry.S [hunk 1]
```diff
@@ -24,20 +24,14 @@ IMAGE_BASE:
 /*    * XSAVE header */
 .align 64
 .Lxsave_clear:
-.org .+24
-.Lxsave_mxcsr:
-    .int 0
+    .byte 0x7f
+    .byte 0x03
```
Comment (jethrogb) [inline src/libstd/sys/sgx/abi/entry.S:28]: This value is the default, you shouldn't need to provide this.

File: src/libstd/sys/sgx/abi/entry.S [hunk 2]
```diff
@@ -24,20 +24,14 @@ IMAGE_BASE:
 /*    * XSAVE header */
 .align 64
 .Lxsave_clear:
-.org .+24
-.Lxsave_mxcsr:
```
Comment (jethrogb) [inline src/libstd/sys/sgx/abi/entry.S:28]: Please maintain this label so it's clear which value you're setting

File: src/libstd/sys/sgx/abi/entry.S [hunk 3]
```diff
@@ -24,20 +24,14 @@ IMAGE_BASE:
...
-    .int 0
+    .byte 0x7f
+    .byte 0x03
+.org .+22
+    .byte 0x80
+    .byte 0x1f
```
Comment (jethrogb) [inline src/libstd/sys/sgx/abi/entry.S:31]: Please use `.short` to provide 16-bit constant

**Pull request merged**